### DISCUSSION TITLE
How to read Contract without connecting?

### DISCUSSION THREAD

**Discussion by SaadNiaziDev** [wevm/wagmi#1550] [Q&A]
How can we replicate this in wagmi + web3modal?
For instance if i want to call multiple functions of contract to show data to user like totalSupply,coinName, decimals etc. without even user connecting.
Like this web will work even on non-dapp browsers ==> https://app.safuu.com/#/dashboard

( old method )
let web3 = new Web3( window.etheruem || https://data-seed-prebsc-1-s1.binance.org:8545/) 
web3.contract.methods.anyFunction( ).call()      

i cant figure how to call functions with wagmi without conneting

**Comment by tmm** [comment]
Follow the ["Getting Started"](https://wagmi.sh/core/getting-started) instructions for `@wagmi/core` (`wagmi` if you are using React and want to use Hooks for data fetching), then use [`readContracts`](https://wagmi.sh/core/actions/readContracts) (or [`useContractReads`](https://wagmi.sh/react/hooks/useContractReads) for React):

```ts
import { readContracts } from '@wagmi/core'

// Set up wagmi Client
// …

const contractConfig = {
  abi: […] as const,
  address: '0x…',
}
const result = await readContracts({
  contracts: [
    { ...contractConfig, functionName: 'totalSupply' },
    { ...contractConfig, functionName: 'coinName' },
    { ...contractConfig, functionName: 'decimals' },
  ],
})
```

**Reply by SaadNiaziDev** [reply]
would this work even if i am using non-dapp browser or not connected to any wallet?

**Reply by tmm** [reply]
Yes, all calls in wagmi use the `Client`'s RPC [`provider`](https://wagmi.sh/react/client#provider).
